Packet++/src/WireGuardLayer.cpp

```diff
@@ -0,0 +1,324 @@
+#define LOG_MODULE PacketLogModuleWireGuardLayer
+
+#include "UdpLayer.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "WireGuardLayer.h"
+#include "Logger.h"
+#include <sstream>
+#include <iomanip>
+#include <cstring>
+
+namespace pcpp
+{
+	void WireGuardLayer::parseNextLayer()
+	{
+		size_t headerLen = getHeaderLen();
+		if (m_DataLen <= headerLen || headerLen == 0)
+			return;
+		m_NextLayer = WireGuardLayer::parseWireGuardLayer(m_Data, m_DataLen, this, m_Packet);
+	}
+
+	WireGuardLayer* WireGuardLayer::parseWireGuardLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+	{
+		if (dataLen < sizeof(WireGuardLayer::wg_common_header))
+			return nullptr;
+		wg_common_header* wgHeader = reinterpret_cast<wg_common_header*>(data);
```

Comment:
If we have a method like `getMessageType()` that returns `WireGuardMessageType`, maybe we should use it here?

Reply:
This function is static, so it cannot call the protected function getMessageType(). Therefore, getMessageType is not used in this function.